test/run_pass/test_net_http_request_response.js

```diff
@@ -0,0 +1,206 @@
+var net = require('net');
+
+// Messages for further requests.
+var message = 'Hello IoT.js';
+
+// Options for further requests.
+var options = {
+  method: 'POST',
+  port: 3005,
+  path: '/',
+  headers : {'Content-Length': message.length}
+};
+
+
+var server0 = http.createServer(function(request, response) {
+  response.end('default');
+});
+server0.listen(80, 5);
+
+// Simple get request without options parameter.
+// Every options should be the default:
+// the default port is 80 and the default method is GET.
+var request0 = http.get(function(response) {
+  var str = '';
+
+  response.on('data', function(chunk) {
+    str += chunk;
+  });
+
+  response.on('end', function() {
+    assert.equal(str, 'default');
+    server0.close();
+  });
+});
+
+
+var server1 = http.createServer(function(request, response) {
+  var str = '';
+
+  request.on('data', function (chunk) {
+    str += chunk.toString();
+  });
+
+  request.on('end', function() {
+    response.end(str);
+  });
+});
+server1.listen(3005, 5);
+
+// Simple request with valid utf-8 message.
+var request1 = http.request(options, function(response) {
+  var str = '';
+
+  response.on('data', function(chunk) {
+    str += chunk.toString();
+  });
+
+  response.on('end', function() {
+    assert.equal(str, message);
+    server1.close();
+  });
+});
+request1.end(message, 'utf-8');
+
+
+var server2 = http.createServer(function(request, response) {
+  response.end();
+});
+server2.listen(3006, 5);
+
+// Simple request with end callback.
+var request2Finished = false;
+options.port = 3006;
+var request2 = http.request(options);
+request2.end(message, function() {
+  server2.close();
+  request2Finished = true;
+});
+
+// Call the request2 end again to test the finish state.
+request2.end(message, function() {
+  // This clabback should never be called.
+  assert.equal(request2Finished, false);
+});
+
+
+var server3 = http.createServer(function(request, response) {
+  var str = '';
+
+  request.on('data', function(chunk) {
+    str += chunk;
+  });
+
+  request.on('end', function() {
+    assert.equal(str, message);
+    response.end('processed');
+  });
+});
+server3.listen(3007, 5);
+
+// Simple request with buffer chunk as message parameter.
+options.port = 3007;
+var request3 = http.request(options, function(response) {
+  var str = '';
+
+  response.on('data', function(chunk) {
+    str += chunk;
+  });
+
+  response.on('end', function() {
+    assert.equal(str, 'processed');
+    server3.close();
+  });
+});
+request3.end(new Buffer(message));
+
+
+// This test is to make sure that when the HTTP server
+// responds to a HEAD request, it does not send any body.
+var server4 = http.createServer(function(request, response) {
+  response.writeHead(200);
+  response.end();
+});
+server4.listen(3008, 5);
+
+var request4 = http.request({
+  method: 'HEAD',
+  port: 3008,
+  path: '/'
+}, function(response) {
+  response.on('end', function() {
+    assert.equal(response.statusCode, 200);
```

Comment:
Please clearly use `assert` on what you check for each test. Since this compares the status code, it seems to assume that the test needs any data from server. I suggest the below pattern. The same problem is found elsewhere in this code.

```
var isTestCase1Called = false;
var request1 = http.get(function(res) {
  ....
  res.on('end', function () {
      isTestCase1Called = true;
  });
  ....
});


process.on('exit', function () {
  assert.equal(isTestCase1Called, false);
});
``